src/components/menu/menu.component.js

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import React, { useRef, useCallback, useState } from 'react';
 import PropTypes from 'prop-types';
 import { StyledMenuWrapper, StyledMenuItemsWrapper, StyledMenuItem } from './menu.style';
+import Events from '../../utils/helpers/events';
 
 const Menu = ({ menuType = 'light', children }) => {
+  const menuItemsRefs = useRef(React.Children.map(children, child => child.ref || React.createRef()));
+  const actualFocusedItemIndex = useRef();
+  const [isOpenSubmenu, setIsOpenSubmenu] = useState(null);
+  const setFocusToElement = useCallback((event, index) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    actualFocusedItemIndex.current = index;
+    const isFocusedOnLastMenuItem = index === menuItemsRefs.current.length;
+    const isFocusedOnFirstMenuItem = index === -1;
+    if (isFocusedOnLastMenuItem) {
+      actualFocusedItemIndex.current = 0;
+    } else if (isFocusedOnFirstMenuItem) {
+      actualFocusedItemIndex.current = menuItemsRefs.current.length - 1;
+    }
+
+    menuItemsRefs.current[actualFocusedItemIndex.current].current.focus();
+  }, []);
+
+  const handleKeyDown = useCallback((event, index) => {
+    if (Events.isRightKey(event)) {
+      event.preventDefault();
+      setFocusToElement(event, index + 1);
+    }
+
+    if (Events.isLeftKey(event)) {
+      event.preventDefault();
+      setFocusToElement(event, index - 1);
+    }
+
+    if (!isOpenSubmenu) {
```

Comment:
Is this condition necessary here?

Reply:
Yes, we need this conditional because our MenuItem is in use as a MenuItem adn SubmenuItem at the same time

Reply:
I _think_ @mkrds is suggesting this because the event will already be handled, we're probably just missing a `e.stopPropagation()` in the sub-`MenuItem`.